Behavior of non-versioned and versioned cluster tasks (#45917)
openshift/openshift-docs: modules/op-disabling-non-versioned-and-versioned-cluster-tasks.adoc

// This module is part of the following assembly:
//
// *cicd/pipelines/managing-nonversioned-and-versioned-cluster-tasks.adoc
:_content-type: CONCEPT
[id="disabling-non-versioned-and-versioned-cluster-tasks_{context}"]
= Disabling non-versioned and versioned cluster tasks

> To disable the cluster tasks installed with the {pipelines-shortname} Operator, cluster administrators can set `spec.addon.params: [Name: clusterTasks, Value: "false"]` in the `TektonConfig` custom resource definition (CRD). 

Reviewer: You could make this into an abstract:
```suggestion
As a cluster administrator, you can disable cluster tasks that were installed by the {pipelines-shortname} Operator.
```

> Applying such a `TektonConfig` CRD immediately remove all non-versioned variants of all cluster tasks from the cluster. To reinstall the non-versioned cluster tasks, set the `value` field to `true`.

Reviewer: ```suggestion
When you disable cluster tasks, the Operator removes all the non-versioned cluster tasks and only the latest version of the versioned cluster tasks from the cluster.
```

> However, the versioned cluster tasks are removed based on the following criteria: 

Reviewer: ```suggestion
```

> * The versioned cluster tasks installed during the most recent Operator installation or upgrade, are removed along with the non-versioned variants. 

Reviewer: ```suggestion
```

> * The versioned cluster tasks installed by an older Operator version are retained. To remove them, uninstall the current Operator and install the old Operator version responsible for installing the old versioned cluster tasks.

Reviewer 2: ```suggestion
* The versioned cluster tasks installed by an older Operator version are retained. They are removed them only when the operator is uninstalled or when they are directly manually deleted.
```

notes:
- there is no need to install an older version of operator in any scenario.
- we can also mention, if an versioned clusterTask from a older version (eg: builah-v-1-6) is directly deleted (`oc delete clustertask buildah-v-1-7`), then there is no way to bring it back. The latest version of operator can only create (re-create) versioned (and non-versioned ) clusterTasks based on latest state of the ClusterTasks.

Reviewer: ```suggestion
* The versioned cluster tasks installed by an older Operator version are retained. To remove them, uninstall the current Operator and install the old Operator version responsible for installing the old versioned cluster tasks.

.Procedure

*  In the `TektonConfig` custom resource definition (CRD), set the `clusterTasks` parameter in `spec.addon.params` to `false`.
```

Reviewer: If you provide the example, you can shorten the step to:
```
. To delete all non-versioned cluster tasks and latest versioned cluster tasks, set the `clusterTasks` parameter to `false`:

Reviewer: Please squash all commits to 1 commit.
